### PR TITLE
Add pylint to Lint action

### PR DESCRIPTION
**Pull request by ptrendx**

### CODE REVIEW

**Comment by ptrendx** [comment]
@ksivaman - before you ask - the problem with the additional pylint issues seemed to be connected with lack of pyTorch installed in the image -> pylint could not resolve the `torch.nn.Module` base class and so had issues with e.g. only 1 function being exposed by the class.

File: transformer_engine/pytorch/distributed.py [hunk 1]
```diff
@@ -218,6 +220,7 @@ def forward(
     def backward(
         ctx, *args: Tuple[torch.Tensor, ...]
```
Comment (ksivaman) [inline transformer_engine/pytorch/distributed.py:221]:
```suggestion
        ctx, *args: Tuple[Union[torch.Tensor, None], ...]
```

**Review by ksivaman — APPROVED** [review]
LGTM

**Pull request merged**